### PR TITLE
1.x Fix batch request/response encoding for the new Jackson codec.

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java [hunk 1]
```diff
@@ -238,6 +244,65 @@ public void call(EncoderWrapper encodingCodec, DecoderWrapper decodingCodec) thr
...
+            public void call(EncoderWrapper encodingCodec, DecoderWrapper decodingCodec) throws IOException {
+                String encodedString = encodingCodec.encode(replicationList);
+
+                ReplicationList decodedValue = decodingCodec.decode(encodedString, ReplicationList.class);
+                assertThat(decodedValue.getReplicationList().size(), is(equalTo(1)));
+            }
+        };
+
+        // In replication channel we use JSON only
+        List<CodecWrapper> jsonCodes = Arrays.asList(
```
Comment (qiangdavidliu) [inline eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java]: what about the reverse (encoded with JacksonJson and decoded with Legacy)?

Comment (tbak) [inline eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java]: The test covers all combinations:
Encoding ReplicationList using {JacksonJson,JacksonJson}
Encoding ReplicationList using {JacksonJson,LegacyJacksonJson}
Encoding ReplicationList using {LegacyJacksonJson,JacksonJson}
Encoding ReplicationList using {LegacyJacksonJson,LegacyJacksonJson}

Encoding ReplicationListResponse using {JacksonJson,JacksonJson}
Encoding ReplicationListResponse using {JacksonJson,LegacyJacksonJson}
Encoding ReplicationListResponse using {LegacyJacksonJson,JacksonJson}
Encoding ReplicationListResponse using {LegacyJacksonJson,LegacyJacksonJson}

Comment (qiangdavidliu) [inline eureka-client/src/test/java/com/netflix/discovery/converters/EurekaCodecCompatibilityTest.java]: Ahh yes, thanks. I misread.

**Comment by qiangdavidliu** [comment]
Ship it.

**Pull request merged**